app/models/nearby_facility.rb

```diff
@@ -48,6 +59,32 @@ def make_linestring(polygon)
       polygon.map { |point| "#{point[1]} #{point[0]}" }.join(',')
     end
 
+    def parse_isochrone(response_json)
+      response_json.try(:[], 'resourceSets')
+                   .try(:first)
+                   .try(:[], 'resources')
+                   .try(:first)
+                   .try(:[], 'polygons')
+                   .try(:first)
+                   .try(:[], 'coordinates')
+                   .try(:first)
+    end
+
+    def handle_bing_errors(body, headers)
+      response_body = JSON.parse(body)
```

Comment:
We are now parsing `response.body` twice, which would normally make me want to refactor a little. Except in this case we know this isn't going to be a long-lived implementation.